### PR TITLE
Replace jscoverage with karma-coverage

### PR DESCRIPTION
**Pull request by mramato**
1. Replace our ancient, windows-only jscoverage implementation with karma-coverage, which works on all platforms and does not require a separate instrumentation step. You now run coverage by simply typing `npm run coverage`. This also provides improved results, such as branch coverage, which the old system did not.
2. Updated Testing Guide
3. Remove old instrument link from index.html
4. Remove jscoverage and related code.

__Current stats__
90.38% Statements 81603/90285
81.91% Branches 27629/33729
90.95% Functions 9010/9906
90.38% Lines 80611/89193

these numbers were lower than I expected, but `Source/Workers` is uncovered by unit tests (at least directly), we should probably add some tests for them and it would get back above 93% pretty easily. I didn't want to lie to ourselves by simple excluding them.

For anyone wondering why we stayed on the old system for so long, I actually made this change ~3 years ago but `karma-coverage` fixed a showstopping bug ~8 days ago and that finally allowed us to use it.

### CODE REVIEW

**Comment by mramato** [comment]
@OmarShehata maybe you can take a look at this? The change itself is pretty straightforward.

**Comment by OmarShehata** [comment]
This is wonderful! We can now see coverage results in the browser for every PR (but with WebGL stubs):

http://cesium-dev.s3-website-us-east-1.amazonaws.com/cesium/coverage/Build/Coverage/index.html

And it would be easy to compare against master (which should get built in a similar URL right?) It's also pretty easy to generate coverage for just a single class as described above. The updated doc looks good.

Some recommendations/questions:

* Why was `npm run clean` removed in .travis.yml?
* We no longer need this `Instrumented` folder in the filesToClean list right? https://github.com/AnalyticalGraphicsInc/cesium/blob/cb496c6271694c4e6a11f446d34d37e45185b6b5/gulpfile.js#L70
* The only thing is the command line options for `npm run coverage` aren't really documented anywhere (so you'd have to read the code to find out `--browsers` takes a comma separated list) but maybe that's OK.

File: gulpfile.js [hunk 1]
```diff
@@ -663,6 +653,63 @@ function setStatus(state, targetUrl, description, context) {
...
+            args: [undefined, undefined, undefined, webglStub, undefined]
+        }
+    }, function(e) {
+        if (!process.env.TRAVIS) {
+            if (browsers.length === 1) {
+                open('Build/Coverage/index.html');
+            } else {
+                var dirs = fs.readdirSync('Build/Coverage');
+                dirs.forEach(function(dir) {
+                    open('Build/Coverage/' + dir + '/index.html');
```
[the remark this answers is not in the document]
Comment (mramato) [reply gulpfile.js]: Yeah, this was me trying to give us a well-known location for index.html and the deployment scripts. I may just punt and have coverage always be a single browser.  It's really uncommon to need to run coverage on multiple browsers at once, and in most cases the results are identical

**Comment by mramato** [comment]
> * Why was `npm run clean` removed in .travis.yml?

Because otherwise the coverage results get deleted and wouldn't be deployed.  I'm actually the one that added the clean (4 years ago apparently) and I think it was to ensure no extra files from earlier in the process were left around when makeZipFile was run, but I'm pretty confident it's not needed ad the resulting zips are the same, so it's safe to remove (so that coverage gets deployed).  Coverage itself is not in the zip, just on AWS.

> * We no longer need this `Instrumented` folder in the filesToClean list right?

No, good catch. There  are a few places that need to be updated.

> * The only thing is the command line options for `npm run coverage` aren't really documented anywhere (so you'd have to read the code to find out `--browsers` takes a comma separated list) but maybe that's OK.

This is true for a lot of our scripts. Could be a good code sprint item to clean things up but it certainly hasn't been an issue so far.

**Comment by mramato** [comment]
@OmarShehata it's over-engineered, but we now auto-generate an index.html with a list of browsers with coverage results.  So it works with multiple browsers without any kind of weird folder behavior.

This is ready.

**Comment by mramato** [comment]
Bump

**Comment by OmarShehata** [comment]
I like this - this works pretty well now!

**Pull request merged**